File: TD/TD4/Code/3.cpp
/* Effectuer la division euclidienne de deux entiers */
#include <iostream>

/* Procédure */

void divisEuclid(int a, int b, int &q, int &r)
{
  q = 0;
  r = a;
  
  while (r >= b) {
    q += 1;
    r -= b;
  }
}

/* Utilisation */

int main()
{
  int a;
  int b;
  int q;
  int r;
  
  std::cout << "Donne moi une valeur : " << std::flush;
  std::cin >> a;
  std::cout << "Donne moi une valeur : " << std::flush;
  std::cin >> b;
  
  divisEuclid(a, b, q, r);
  
  std::cout << "La division euclidienne de ces deux valeurs donne un quotient de " << q << " et un reste de " << r << std::endl;
  
  return 0;
}

File: TD/TD4/Code/Python/3.py
#!/usr/bin/python
# Effectuer la division euclidienne de deux entiers


def division(foo, bar):
    q = 0  # Le quotient de division
    r = foo  # LE reste de la division

    while r >= bar:
        q += 1
        r -= bar

    # I ❤ Python #
    # q = foo // bar #
    # r = foo % bar #

    return q, r


print('Donne moi deux valeurs : ', end='')
foo = int(input())  # La valeur à diviser (numérateur)
bar = int(input())  # Le dénominateur

q, r = division(foo, bar)

print('Le quotient de la division est :', q,
      'et le reste est :', r)
